addons/point_of_sale/static/src/js/Chrome.js

```diff
@@ -1,120 +1,141 @@
+
         super.setup();
         useExternalListener(window, "beforeunload", this._onBeforeUnload);
         useListener("show-main-screen", this.__showScreen);
-        useListener("toggle-debug-widget", debounce(this._toggleDebugWidget, 100));
-        useListener("toggle-mobile-searchbar", this._toggleMobileSearchBar);
         useListener("show-temp-screen", this.__showTempScreen);
         useListener("close-temp-screen", this.__closeTempScreen);
         useListener("close-pos", this._closePos);
         useListener("loading-skip-callback", () => this.env.proxy.stop_searching());
-        useListener("play-sound", this._onPlaySound);
+        const sound = useService("sound");
+        useListener("play-sound", ({ detail: name }) => sound.play(name));
         useListener("set-sync-status", this._onSetSyncStatus);
         useListener("show-notification", this._onShowNotification);
         useListener("close-notification", this._onCloseNotification);
         useListener("connect-to-proxy", this.connect_to_proxy);
         useBus(this.env.posbus, "start-cash-control", this.openCashControl);
         NumberBuffer.activate();
 
-        this.state = useState({
-            uiState: "LOADING", // 'LOADING' | 'READY' | 'CLOSING'
-            debugWidgetIsShown: true,
-            mobileSearchBarIsShown: false,
-            hasBigScrollBars: false,
-            sound: { src: null },
-            notification: {
-                isShown: false,
-                message: "",
-                duration: 2000,
-            },
-            loadingSkipButtonIsShown: false,
-        });
+        this.state = usePos();
 
-        this.mainScreen = useState({ name: null, component: null });
+        this.mainScreen = this.state.mainScreen;
         this.mainScreenProps = {};
 
         this.tempScreen = useState({ isShown: false, name: null, component: null });
         this.tempScreenProps = {};
 
-        this.progressbar = useRef("progressbar");
-
-        this.previous_touch_y_coordinate = -1;
-
-        const pos = reactive(
-            this.env.pos,
-            batched(() => this.render(true))
-        );
-        useSubEnv({ pos });
+        useSubEnv({
+            pos: reactive(
+                this.env.pos,
+                batched(() => this.render(true)) // FIXME POSREF remove render(true)
+            ),
+        });
 
         onMounted(() => {
             // remove default webclient handlers that induce click delay
+            // FIXME POSREF if these handlers shouldn't be there we should not load the files that add them.
             $(document).off();
             $(window).off();
             $("html").off();
             $("body").off();
         });
 
         onWillDestroy(() => {
-            this.env.pos.destroy();
-        });
-
-        onError((error) => {
-            // error is an OwlError object.
-            console.error(error.cause);
+            try {
+                // FIXME POSREF is this needed?
```

Comment:
My question was more related to the fact that when the Chrome is destroyed it means that the entire page is about to be reloaded and any stray resources should be cleaned up automatically by the browser. It might be useful for tests though.